magenta-lib/src/main/scala/magenta/tasks/FastlyTasks.scala

```diff
@@ -30,43 +31,73 @@ case class UpdateFastlyConfig(s3Package: S3Path)(implicit
   implicit val ec: ExecutionContextExecutorService =
     ExecutionContext.fromExecutorService(Executors.newFixedThreadPool(10))
 
-  override def execute(
+  private def _execute(
+      client: FastlyApiClient,
       resources: DeploymentResources,
       stopFlag: => Boolean
-  ): Unit = {
-    FastlyApiClientProvider.get(keyRing).foreach { client =>
-      val activeVersionNumber =
+  ): Try[Int] = {
+    for {
+      activeVersionNumber <- Try(
         getActiveVersionNumber(client, resources.reporter, stopFlag)
-      val nextVersionNumber =
+      )
+      nextVersionNumber <- Try(
         clone(activeVersionNumber, client, resources.reporter, stopFlag)
-
-      deleteAllVclFilesFrom(
-        nextVersionNumber,
-        client,
-        resources.reporter,
-        stopFlag
       )
-
-      uploadNewVclFilesTo(
-        nextVersionNumber,
-        s3Package,
-        client,
-        resources.reporter,
-        stopFlag
+      _ <- Try(
+        deleteAllVclFilesFrom(
+          nextVersionNumber,
+          client,
+          resources.reporter,
+          stopFlag
+        )
       )
-
-      commentVersion(
-        nextVersionNumber,
-        client,
-        resources.reporter,
-        parameters,
-        stopFlag
+      _ <-
+        Try(
+          uploadNewVclFilesTo(
+            nextVersionNumber,
+            s3Package,
+            client,
+            resources.reporter,
+            stopFlag
+          )
+        )
+      _ <- Try(
+        commentVersion(
+          nextVersionNumber,
+          client,
+          resources.reporter,
+          parameters,
+          stopFlag
+        )
       )
+      _ <- Try(
+        activateVersion(
+          nextVersionNumber,
+          client,
+          resources.reporter,
+          stopFlag
+        )
+      )
+    } yield nextVersionNumber
+  }
 
-      activateVersion(nextVersionNumber, client, resources.reporter, stopFlag)
-
-      resources.reporter
-        .info(s"Fastly version $nextVersionNumber is now active")
+  override def execute(
+      resources: DeploymentResources,
+      stopFlag: => Boolean
+  ): Unit = {
+    FastlyApiClientProvider.get(keyRing) match {
+      case None =>
+        resources.reporter.fail("Failed to fetch Fastly API credentials")
+      case Some(client) =>
+        _execute(client, resources, stopFlag) match {
+          case Success(nextVersionNumber) =>
+            resources.reporter
+              .info(s"Fastly version $nextVersionNumber is now active")
+          case Failure(err) =>
+            resources.reporter.fail(
+              s"$err. Your API key may be invalid or it may have expired"
+            )
+        }
```

Comment:
I think this might be slightly easier to read:
```suggestion
    override def execute(
      resources: DeploymentResources,
      stopFlag: => Boolean
  ): Unit = {
    FastlyApiClientProvider
      .get(keyRing)
      .map(_execute(_, resources, stopFlag)) match {
      case Some(Success(nextVersionNumber)) =>
        resources.reporter
          .info(s"Fastly version $nextVersionNumber is now active")
      case Some(Failure(err)) =>
        resources.reporter.fail(
          s"$err. Your API key may be invalid or it may have expired"
        )
      case None =>
        resources.reporter.fail("Failed to fetch Fastly API credentials")
    }
  }
```